sdk/test-utils/recorder-new/test/testProxyClient.spec.ts

```diff
@@ -7,33 +7,32 @@ import {
   PipelineRequest,
   PipelineResponse
 } from "@azure/core-rest-pipeline";
-import { env } from "@azure-tools/test-recorder";
 import { expect } from "chai";
-import { TestProxyHttpClient } from "../src";
-import { RecorderError, RecordingStateManager } from "../src/utils/utils";
+import { env, Recorder } from "../src";
+import { getTestMode, isLiveMode, RecorderError, RecordingStateManager } from "../src/utils/utils";
 
 const testRedirectedRequest = (
-  client: TestProxyHttpClient,
+  client: Recorder,
   makeRequest: () => PipelineRequest,
   expectedModification: (req: PipelineRequest) => PipelineRequest
 ) => {
   const redirectedRequest = makeRequest();
-  client.redirectRequest(redirectedRequest);
+  client["redirectRequest"](redirectedRequest);
```

Comment:
I'm just confused about why I'm seeing a lot of properties being accessed as elements like this and not using the dot operator.

Reply:
they are not public since I don't expect users to use them, they are private now.

Reply:
I had absolutely no idea until right now that TypeScript would let you just... get away with this. Just changing from property access to a string indexer is enough to defeat the `private` modifier?

It does still feel a little strange to me, I guess because I've never seen it before, but I can understand why you want to reach into the private fields just for the purpose of testing the recorder's internals.

Reply:
I feel like I've been asleep in a hibernation tube for a hundred years and I've just woken up and the first thing anyone has said to me is "accessing an object member by string accessor is not the same thing as by property name in TypeScript" and I just can't come to terms with this new reality.

Reply:
Yes, also this is not being added in src code, only as part of the tests.
If typescript ever changes its behavior we might have to change such instances accordingly.

This is a pattern I learned or we discovered while writing service-bus tests.